web/src/vdom.ts

```diff
@@ -2,7 +2,43 @@ import _ from "lodash";
 
 import * as blueslip from "./blueslip";
 
-export function eq_array(a, b, eq) {
+// export type
+export type VDomNode = {
+    key: string;
+    eq: (other: VDomNode) => boolean;
+    render: () => string;
+    conversation: Conversation;
```

Comment:
This module is supposed to be generic and shouldn’t know anything about details like “conversation” that are specific to only one of its users. For example, it’s also used in `topic_list.js`.

Reply:
I think we should use type of  `Record<string, unknown>`

Reply:
Ideally we would use [generic types](https://www.typescriptlang.org/docs/handbook/2/generics.html); that’s what they’re for.

Reply:
I've made the required changes, ie `type VDomNode` is now generic

Reply:
You’ve missed the whole point of making this generic. You’ve added a `T` parameter that isn’t actually used anywhere, and you haven’t addressed the requirement that the vdom module shouldn’t know anything about details like “conversation”.

Reply:
No—there should be **no references to anything called “conversation”** inside `vdom.ts`.

Reply:
No, you haven’t. You’re still adding references to “conversation” in `vdom.ts`.

Reply:
fixed it @andersk